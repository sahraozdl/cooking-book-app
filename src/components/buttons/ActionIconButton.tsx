import React from 'react';

interface ActionIconButtonProps {
  onClick: () => void;
  label: string;
  icon: React.ReactNode;
  activeIcon?: React.ReactNode;
  active?: boolean;
  className?: string;
  count?: number;
}

export default function ActionIconButton({
  onClick,
  label,
  icon,
  activeIcon,
  active = false,
  count,
  className = '',
}: ActionIconButtonProps) {
  return (
    <button
      onClick={onClick}
      className={`flex items-center text-sm gap-1 transition focus:outline-orange-500 focus:ring-2 focus:ring-orange-400 rounded-full px-2 py-1 hover:animate-pulse cursor-pointer hover:scale-125
        ${active ? 'text-orange-600 bg-orange-200' : 'text-gray-400 border border-orange-300'} ${className}`}
      aria-label={label}
    >
      <span className="text-lg">{active ? (activeIcon ?? icon) : icon}</span>
      <span className="text-xs">{count}</span>
    </button>
  );
}
