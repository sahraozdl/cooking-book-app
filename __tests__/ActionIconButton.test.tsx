import { render, screen, fireEvent } from '@testing-library/react';
import ActionIconButton from '@/components/buttons/ActionIconButton';
import { HeartIcon } from '@phosphor-icons/react';

describe('ActionIconButton', () => {
  it('renders the button with icon and label', () => {
    render(
      <ActionIconButton
        onClick={() => {}}
        label="Like"
        icon={<HeartIcon data-testid="icon" size={16} />}
        count={5}
      />
    );

    expect(screen.getByLabelText('Like')).toBeInTheDocument();
    expect(screen.getByText('5')).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });

  it('renders activeIcon when active', () => {
    render(
      <ActionIconButton
        onClick={() => {}}
        label="Like"
        icon={<HeartIcon data-testid="icon" size={16} />}
        activeIcon={<HeartIcon data-testid="active-icon" weight="fill" size={16} />}
        active
      />
    );

    expect(screen.getByTestId('active-icon')).toBeInTheDocument();
  });

  it('handles click events', () => {
    const handleClick = jest.fn();
    render(<ActionIconButton onClick={handleClick} label="Like" icon={<HeartIcon size={16} />} />);

    fireEvent.click(screen.getByRole('button', { name: /Like/i }));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ActionIconButton onClick={() => {}} label="Like" icon={<HeartIcon size={16} />} count={3} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
