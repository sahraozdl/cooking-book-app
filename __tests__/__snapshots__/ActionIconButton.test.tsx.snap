// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ActionIconButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Like"
    class="flex items-center text-sm gap-1 transition focus:outline-orange-500 focus:ring-2 focus:ring-orange-400 rounded-full px-2 py-1 hover:animate-pulse cursor-pointer hover:scale-125
        text-gray-400 border border-orange-300 "
  >
    <span
      class="text-lg"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 256 256"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M178,40c-20.65,0-38.73,8.88-50,23.89C116.73,48.88,98.65,40,78,40a62.07,62.07,0,0,0-62,62c0,70,103.79,126.66,108.21,129a8,8,0,0,0,7.58,0C136.21,228.66,240,172,240,102A62.07,62.07,0,0,0,178,40ZM128,214.8C109.74,204.16,32,155.69,32,102A46.06,46.06,0,0,1,78,56c19.45,0,35.78,10.36,42.6,27a8,8,0,0,0,14.8,0c6.82-16.67,23.15-27,42.6-27a46.06,46.06,0,0,1,46,46C224,155.61,146.24,204.15,128,214.8Z"
        />
      </svg>
    </span>
    <span
      class="text-xs"
    >
      3
    </span>
  </button>
</DocumentFragment>
`;
